### ISSUE TITLE
Wrong expr simplification?

### ISSUE THREAD

**Issue by naoyam** [NVIDIA/Fuser#2486]
Hit a validation error with the new tensor indexer. To repro, use the branch of #2238 and run `NVFUSER_ENABLE=id_model(producer_index,consumer_index,inline_predicate,vectorize_predicate,unswitch_predicate) NVFUSER_DISABLE=index_hoist,parallel_compile ./bin/nvfuser_tests --gtest_filter="*DoubleBuffering6"`

Here's what I get:

```
 NVFUSER_ENABLE=id_model(producer_index,consumer_index,inline_predicate,vectorize_predicate,unswitch_predicate) NVFUSER_DISABLE=index_hoist,parallel_compile ./bin/nvfuser_tests --gtest_filter="*DoubleBuffering6"
Running main() from /home/nmaruyama/nvfuser/debug2/third_party/googletest/googletest/src/gtest_main.cc
Note: Google Test filter = *DoubleBuffering6
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from DoubleBufferingTest
[ RUN      ] DoubleBufferingTest.DoubleBuffering6
unknown file: Failure
C++ exception with description "aten_output_tensor.allclose( fusion_output_tensor.to(aten_output_tensor.dtype()), tolerance_values.second, tolerance_values.first, true) INTERNAL ASSERT FAILED at "/home/nmaruyama/nvfuser/debug2/tests/cpp/validator.h":142, please report a bug with repro script to NVFuser at https://github.com/NVIDIA/Fuser/issues.

Validation error in output 0 on line 263 in file /home/nmaruyama/nvfuser/debug2/tests/cpp/test_double_buffering.cpp.
  Detected abs error of: 4.56514
    absolute tolerance was set to 1.68222e-06
    and relative tolerance set to 2.23704e-06
Exception raised from testValidate at /home/nmaruyama/nvfuser/debug2/tests/cpp/validator.h:142 (most recent call first):
frame #0: <unknown function> + 0x6b3288 (0x55f2b3ae0288 in ./bin/nvfuser_tests)
frame #1: <unknown function> + 0x6b34ba (0x55f2b3ae04ba in ./bin/nvfuser_tests)
frame #2: <unknown function> + 0xb9371a (0x55f2b3fc071a in ./bin/nvfuser_tests)
frame #3: <unknown function> + 0xb9878e (0x55f2b3fc578e in ./bin/nvfuser_tests)
frame #4: <unknown function> + 0x121b19d (0x55f2b464819d in ./bin/nvfuser_tests)
frame #5: <unknown function> + 0x1202239 (0x55f2b462f239 in ./bin/nvfuser_tests)
frame #6: <unknown function> + 0x1202fe9 (0x55f2b462ffe9 in ./bin/nvfuser_tests)
frame #7: <unknown function> + 0x1203a75 (0x55f2b4630a75 in ./bin/nvfuser_tests)
frame #8: <unknown function> + 0x1212c4d (0x55f2b463fc4d in ./bin/nvfuser_tests)
frame #9: <unknown function> + 0x121bd6d (0x55f2b4648d6d in ./bin/nvfuser_tests)
frame #10: <unknown function> + 0x1212463 (0x55f2b463f463 in ./bin/nvfuser_tests)
frame #11: <unknown function> + 0x11e23eb (0x55f2b460f3eb in ./bin/nvfuser_tests)
frame #12: <unknown function> + 0x29d90 (0x7fa171fd4d90 in /usr/lib/x86_64-linux-gnu/libc.so.6)
frame #13: __libc_start_main + 0x80 (0x7fa171fd4e40 in /usr/lib/x86_64-linux-gnu/libc.so.6)
frame #14: <unknown function> + 0x50f855 (0x55f2b393c855 in ./bin/nvfuser_tests)
" thrown in the test body.

To reproduce: NVFUSER_TEST_RANDOM_SEED=1719536409 NVFUSER_TEST_ATEN_RANDOM_SEED=0 nvfuser_tests --gtest_filter='DoubleBufferingTest.DoubleBuffering6'
[  FAILED  ] DoubleBufferingTest.DoubleBuffering6 (905 ms)
[----------] 1 test from DoubleBufferingTest (905 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (905 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] DoubleBufferingTest.DoubleBuffering6

 1 FAILED TEST
```

Disabling the expr simplification seems to work around the problem. I also disabled index hoisting but it doesn't seem to affect the result.

**Comment by naoyam** [comment]
CC: @zasdfgbnm

**Comment by naoyam** [comment]
That was due to my incorrect Val replacement. It was just exposed with expr simplification. Fixed at f7fbb9d5e5d4d42f895bcd355108123b95829c61. Thanks @zasdfgbnm for the help.
